controllers/authentication/openidconnect_controller.go

```diff
@@ -89,15 +91,28 @@ func (r *OpenIDConnectReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
-	// retrieve the JWKS keySet from the jwksURL endpoint
 	var keySet gooidc.KeySet
-	keySet, err = remoteKeySet(ctx, config.Spec.IssuerURL, config.Spec.CABundle)
-	if err != nil {
-		log.Error(err, "Invalid JWKS KeySet")
 
-		r.handlers.Delete(req.Name)
+	if config.Spec.JWKS.Keys != nil {
+		keySet, err = newStaticKeySet(config.Spec.JWKS.Keys)
+		if err != nil {
+			log.Error(err, "Invalid static JWKS KeySet")
 
-		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			r.handlers.Delete(req.Name)
+
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
This part is the same in both static and remote keys. Simply move this part outside.